### ISSUE TITLE
Since 4.1.0, fallback AWS credential providers below the Cantaloupe config provider cannot be used

### ISSUE THREAD

**Issue by MorrisonCole** [cantaloupe-project/cantaloupe#333]
**TL;DR:** Cantaloupe 4.1+ uses empty AWS credentials if `EnvironmentVariableCredentialsProvider` and `SystemPropertiesCredentialsProvider` fail and the config fallback is not specified. This is because the config fallback accepts empty credentials. This means that `ProfileCredentialsProvider` and `EC2ContainerCredentialsProviderWrapper` are never checked.

**The Problem**
I'm trying to deploy Cantaloupe `4.1.4` on ECS as with a task role, but I cannot set up authentication in the usual way since the `EC2ContainerCredentialsProviderWrapper` is never checked for credentials.

I believe that this is because since 4.1.0 (specifically, [this commit](https://github.com/cantaloupe-project/cantaloupe/commit/8b6ff77911812b434c386f7b50afbee0f98a2b2d#diff-4af9f42ceb870e2e6f21cdf208bd3484)), empty credentials have not been guarded against as [they used to be](https://github.com/cantaloupe-project/cantaloupe/blob/v4.0/src/main/java/edu/illinois/library/cantaloupe/util/AWSClientBuilder.java#L131). Since I want to use `EC2ContainerCredentialsProviderWrapper`, I leave the Cantaloupe config blank but it is used anyway. 🍈

**Comment by MorrisonCole** [comment]
Actually, I'm not sure there is actually a bug here, since if the credentials are not specified they are null, and internally the AWS SDK moves on to the next provider in this case.

I'll close this for now and re-open with more details if I can pin it down.

**Comment by MorrisonCole** [comment]
Finally got a chance to come back and confirm this - it does indeed seem to be the problem. Somehow configuration returns an empty string rather than `null`, although I haven't debugged it fully to make a proper fix.

**Comment by adolski** [comment]
Thanks for reporting this, @MorrisonCole. It looks like it could be fixed by checking for a non-empty `accessKeyIDFromConfig` before [adding the custom AWSCredentialsProvider](https://github.com/cantaloupe-project/cantaloupe/blob/d2471e44a5b3645256cfedd080beafdec7f1ee99/src/main/java/edu/illinois/library/cantaloupe/util/AWSClientBuilder.java#L55)?

**Comment by MorrisonCole** [comment]
@adolski no problem! I think it could be worked around that way for sure. I had a look and think that the root cause is a little less obvious, but had a go at fixing it properly -> #335 😄

**Comment by adolski** [comment]
Thanks for fixing this, @MorrisonCole!
